server/service/targets.go

```diff
@@ -264,3 +354,32 @@ func (svc *Service) CountHostsInTargets(ctx context.Context, queryID *uint, targ
 
 	return &metrics, nil
 }
+
+type countTargetsRequest struct {
+	Selected fleet.HostTargets `json:"selected"`
+	QueryID  *uint             `json:"query_id"`
+}
+
+type countTargetsResponse struct {
+	TargetsCount   uint  `json:"targets_count"`
+	TargetsOnline  uint  `json:"targets_online"`
+	TargetsOffline uint  `json:"targets_offline"`
+	Err            error `json:"error,omitempty"`
+}
+
+func (r countTargetsResponse) error() error { return r.Err }
+
+func countTargetsEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (interface{}, error) {
+	req := request.(*countTargetsRequest)
+
+	counts, err := svc.CountHostsInTargets(ctx, req.QueryID, req.Selected)
```

Comment:
Similar comment here, maybe this should be `countHostsInTargets`? _Technically_ if you target 1 label, 1 team, and 1 host, the amount of targets is 3, even if there are 100 hosts in that label.